src/android/platform.cpp

```diff
@@ -114,4 +114,15 @@ void print(const char* fmt, ...)
     __android_log_vprint(ANDROID_LOG_INFO, "RealmJS", fmt, vl);
     va_end(vl);
 }
+
+std::string get_cpu_arch()
+{
+#define QUOTE(name) #name
+#define STR(name) QUOTE(name)
+#define ABI_NAME STR(REALM_ANDROID_ABI)
+    return ABI_NAME;
+#undef STR
+#undef QUOTE
```

Comment:
This looks a bit suspicious to me 🙂 But I [read a bit more about it](https://gcc.gnu.org/onlinedocs/cpp/Stringizing.html#Stringizing).

However, this could be simplified (including passing in the `REALM_ANDROID_ABI`) by relying on the GCC preprocessor directives:

```suggestion
#if defined(__arm__)
    return "armeabi-v7a";
#elif defined(__aarch64__)
    return "arm64-v8a";
#elif defined(__i386__)
    return "x86";
#elif defined(__x86_64__)
    return "x86_64";
#else
    return "unknown";
#endif
```

This has the benefit that we can control the values to match that of the other platforms.

Reply:
I am an old C hacker so I like the preprocessor 👴🏻 but your suggested is easier to read and doesn't rely on changes in `CMakeList.txt`